dhcp_server/DOCS.md

```diff
@@ -1,34 +1,34 @@
-# Home Assistant Add-on: DHCP server
+# Home Assistant Add-on: DHCP Server
```

Comment:
We generally use [sentence style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization), also [for titles](https://learn.microsoft.com/en-us/style-guide/text-formatting/formatting-titles), as per Microsoft writing style guideline.